seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/source/ClickhouseSourceFactory.java

```diff
@@ -117,11 +183,80 @@ private String modifySQLToLimit1(String sql) {
         return String.format("SELECT * FROM (%s) s LIMIT 1", sql);
     }
 
+    private String generateQuerySql(String sql, String database, String table) {
+        if (StringUtils.isNotEmpty(sql)) {
+            return modifySQLToLimit1(sql);
+        }
+
+        return String.format("SELECT * FROM %s.%s LIMIT 1", database, table);
+    }
+
+    private List<Shard> getClusterShardList(
```

Comment:
I think generate shard list can be move into `ClickhouseSourceSplitEnumerator`. That's the place to generate splits. We need to ensure that responsibilities are clear